src/main/java/com/czertainly/core/service/impl/UserManagementServiceImpl.java

```diff
@@ -126,4 +118,22 @@ public List<RoleDto> getUserRoles(String userUuid) {
     public UserDetailDto removeRole(String userUuid, String roleUuid) {
         return userManagementApiClient.removeRole(userUuid, roleUuid);
     }
+
+    private Certificate addUserCertificate(String certificateUuid, String certificateData) throws CertificateException, NotFoundException {
+
+        Certificate certificate;
+        if (StringUtils.isNotBlank(certificateUuid)) {
+            certificate = certificateService.getCertificateEntity(SecuredUUID.fromString(certificateUuid));
+        } else {
+            X509Certificate x509Cert = CertificateUtil.parseCertificate(certificateData);
+            try {
+                certificate = certificateService.getCertificateEntityBySerial(x509Cert.getSerialNumber().toString(16));
+            } catch (NotFoundException e) {
+                logger.debug("New Certificate uploaded for admin");
```

Comment:
Should be user in the message, not admin

Reply:
It should be. I will update it